### DISCUSSION TITLE
Global State in a webview

### DISCUSSION THREAD

**Discussion by caleb-at-pieces** [microsoft/vscode-discussions#1165] [Extension Development QnA] [answered]
Hello,

I would like to know if there is a quick and simple way to implement global persistence in a webview.

I'm aware of the `getState` and `saveState` callbacks however those two seem to be workspace specific, however I would like to save webview state globally, kind of like `vscode.ConfigurationTarget.Global` while modifying a setting.

My current thoughts are to use the globalState property on the extension context object, I am just wondering if there is a better solution that allows me to not need to have some message passing back and forth with the extension.

**Comment by tamuratak** [comment, marked as answer]
You could try to use `localStorage`. Please refer to

- [Consistent origin for webviews and webview views](https://code.visualstudio.com/updates/v1_73#_consistent-origin-for-webviews-and-webview-views)
- https://github.com/microsoft/vscode/issues/132464#issuecomment-1317800583

> Webviews can use local storage and other web APIs that are partitioned per origin.
> 
> Keep in mind that all instances of a webview will now run on the same origin so if they are using an API such as local storage, be sure to partition any data/state that is document specific per resource. For example, localStorage.setItem('scrollPosition', 100) will set scrollPosition to 100 across all webview instances.

**Comment by abrilurena** [comment]
Hi, @caleb-at-pieces! do you still need help with this?
